src/main/java/seedu/address/model/person/Major.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's major in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMajor(String)}
+ */
+public class Major {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Majors should be at least 2 characters long";
+
+    public static final String VALIDATION_REGEX = "(?<!\\s)[\\p{Alpha} ]{2,}";
```

Comment:
I'm not sure what the `(?<!\\s)` does here. I think it wld be good to document what this regex does.

Reply:
Good catch! Thanks!